Currently, I'm getting this:

```
Recreating tahoe.devstack.postgres ... done
Creating tahoe.devstack.amc        ... done
make[1]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make tahoe.provision
make[1]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
make tahoe.chown
make[2]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
sudo chown anders -R /home/anders/work/tahoe-devstack/devstack/..
[sudo] password for anders: 
make[2]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
cat /home/anders/work/tahoe-devstack/devstack/../devstack/provision-tahoe.py > /home/anders/work/tahoe-devstack/devstack/../src/provision-tahoe.py
make COMMAND='python /edx/src/provision-tahoe.py' tahoe.exec.edxapp
make[2]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
make COMMAND='python /edx/src/provision-tahoe.py' SERVICE=lms tahoe.exec.single
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t edx.devstack.lms  \
	bash -c 'source /edx/app/edxapp/edxapp_env && cd /edx/app/edxapp/edx-platform/ && python /edx/src/provision-tahoe.py'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make COMMAND='python /edx/src/provision-tahoe.py' SERVICE=studio tahoe.exec.single
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t edx.devstack.studio  \
	bash -c 'source /edx/app/edxapp/edxapp_env && cd /edx/app/edxapp/edx-platform/ && python /edx/src/provision-tahoe.py'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make[2]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
rm /home/anders/work/tahoe-devstack/devstack/../src/provision-tahoe.py
make tahoe.restart || true
make[2]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
make lms-restart
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t edx.devstack.lms bash -c 'kill $(ps aux | grep "manage.py lms" | egrep -v "while|grep" | awk "{print \$2}")'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make studio-restart
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t edx.devstack.studio bash -c 'kill $(ps aux | grep "manage.py cms" | egrep -v "while|grep" | awk "{print \$2}")'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make amc-restart
make[3]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -t tahoe.devstack.amc bash -c 'killall5'
docker exec -t tahoe.devstack.amc-frontend bash -c 'killall5'
make[3]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make[2]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make[1]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
make amc.provision
make[1]: Entering directory '/home/anders/work/tahoe-devstack/devstack'
docker exec -it tahoe.devstack.amc python manage.py migrate
python: can't open file 'manage.py': [Errno 2] No such file or directory
tahoe.mk:36: recipe for target 'amc.provision' failed
make[1]: *** [amc.provision] Error 2
make[1]: Leaving directory '/home/anders/work/tahoe-devstack/devstack'
Makefile:62: recipe for target 'dev.provision' failed
make: *** [dev.provision] Error 2
```

but I suspect that it is because of the issue I had with https://github.com/appsembler/amc/pull/185

--- a/Makefile
+++ b/Makefile
@@ -58,7 +58,11 @@ dev.clone: ## Clone service repos to the parent directory
 dev.provision.run: ## Provision all services with local mounted directories
 	DOCKER_COMPOSE_FILES="-f docker-compose.yml -f docker-compose-host.yml" $(WINPTY) bash ./provision.sh
 
-dev.provision: | check-memory dev.clone dev.provision.run stop ## Provision dev environment with all services stopped
+dev.provision: | check-memory dev.clone dev.provision.run ## Provision dev environment with all services stopped
+	make dev.up
+	make tahoe.provision
+	make amc.provision
+	make stop
 
 dev.provision.xqueue: | check-memory dev.provision.xqueue.run stop stop.xqueue
 

--- a/README.rst
+++ b/README.rst
@@ -13,7 +13,6 @@ This project requires **Docker 17.06+ CE**.
     $ cd ~/work/tahoe-hawthorn
     $ git clone https://github.com/appsembler/devstack.git
     $ cd devstack  # Now the `devstack` repo should be on the `hawthorn` branch
-    $ make dev.clone
     $ make dev.provision
     $ make tahoe.up
 
@@ -77,29 +76,15 @@ Currently the commands looks like this:
 .. code::
 
     $ make help | grep -e tahoe -e amc
-      amc.env-file              Removes and uses a fresh copy of the AMC env file
-      amc.migrate               Migrate the AMC database
-      amc.reset                 Removes and re-initialize AMC
-      amc.start.backend         Starts the AMC Django app
-      amc.start.frontend        Starts the AMC Frontend
-      tahoe.amc.oauth-client    Creates the AMC OAuth client in the LMS
-      tahoe.amc.superuser       Creates the devstack admin user
-      tahoe.chown               Fix an annoying docker permission issue in both `edx-platform` and `src`
+      amc.provision             Initializes the AMC
+      tahoe.chown               Fix annoying docker permission issues
       tahoe.envs._delete        Remove settings, in prep for resetting it
-      tahoe.envs.reset          Reset the JSON envs
       tahoe.exec.edxapp         Execute a command in both LMS and Studio (edxapp containers)
       tahoe.exec.single         Execute a command inside a devstack docker container
-      tahoe.figures             Install Figures
-      tahoe.init                Make the devstack more Tahoe'ish
-      tahoe.init.provision-script Execute the `provision-tahoe.py` script in both of LMS and Studio
-      tahoe.install-pip.edxapp  Install a pip package in both of LMS and Studio
-      tahoe.os.exec             Executes operating system-specific commands
-      tahoe.reset.full          Does a full reset for everything known to devstack. Will loose all git and database changes.
-      tahoe.reset.light         Resets the Tahoe settings including a fresh theme copy and new environment files.
+      tahoe.provision           Make the devstack more Tahoe'ish
       tahoe.restart             Restarts both of LMS and Studio python processes while keeping the same container
-      tahoe.theme.compile       Compile the static assets of the theme
-      tahoe.theme.reset         Removes and re-clone the theme with Tahoe branches
       tahoe.up                  Run the devstack with proper Tahoe settings, use instead of `$ make dev.up`
+
 
 If something goes wrong, check the rest of this README for additional details.
 

--- a/docker-compose-host.yml
+++ b/docker-compose-host.yml
@@ -21,6 +21,9 @@ services:
       - ${DEVSTACK_WORKSPACE}/edx-platform:/edx/app/edxapp/edx-platform:cached
       - edxapp_node_modules:/edx/app/edxapp/edx-platform/node_modules
       - ${DEVSTACK_WORKSPACE}/src:/edx/src:cached
+      - ${DEVSTACK_WORKSPACE}/customer_themes:/edx/var/edxapp/themes/customer_themes:cached
+      - ${DEVSTACK_WORKSPACE}/edx-theme-codebase:/edx/var/edxapp/themes/edx-theme-codebase:cached
+      - ${DEVSTACK_WORKSPACE}/edx-theme-customers:/edx/var/edxapp/themes/edx-theme-codebase/customer_specific:cached
   edx_notes_api:
     volumes:
       - ${DEVSTACK_WORKSPACE}/edx-notes-api:/edx/app/edx_notes_api/edx_notes_api:cached
@@ -30,10 +33,19 @@ services:
       - ${DEVSTACK_WORKSPACE}/edx-platform:/edx/app/edxapp/edx-platform:cached
       - edxapp_node_modules:/edx/app/edxapp/edx-platform/node_modules
       - ${DEVSTACK_WORKSPACE}/src:/edx/src:cached
+      - ${DEVSTACK_WORKSPACE}/customer_themes:/edx/var/edxapp/themes/customer_themes:cached
+      - ${DEVSTACK_WORKSPACE}/edx-theme-codebase:/edx/var/edxapp/themes/edx-theme-codebase:cached
+      - ${DEVSTACK_WORKSPACE}/edx-theme-customers:/edx/var/edxapp/themes/edx-theme-codebase/customer_specific:cached
   forum:
     volumes:
       - ${DEVSTACK_WORKSPACE}/cs_comments_service:/edx/app/forum/cs_comments_service:cached
-      
+  amc:
+    volumes:
+      - ${DEVSTACK_WORKSPACE}/amc:/opt/amc:cached
+  amc-frontend:
+    volumes:
+      - ${DEVSTACK_WORKSPACE}/amc:/opt/amc:cached
+
 volumes:
   credentials_node_modules:
   discovery_node_modules:

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -225,6 +225,50 @@ services:
     volumes:
       - redis_data:/data
 
+  amc:
+    command: bash -c 'while true; do python manage.py runserver 0.0.0.0:19000; sleep 2; done'
+    container_name: tahoe.devstack.amc
+    depends_on:
+      - redis
+      - lms
+      - postgres
+    # Allows attachment to the LMS service using 'docker attach <containerID>'.
+    stdin_open: true
+    tty: true
+    environment:
+      DJANGO_DEBUG: "true"
+      DJANGO_CELERY_ALWAYS_EAGER: "true"
+      SSL_ENABLED: "false"
+      DATABASE_URL: "postgres://postgres:postgres@tahoe.devstack.postgres:5432/postgres"
+      BROKER_URL: "redis://tahoe.devstack.redis:6379/1"
+      LMS_BASE_URL: "http://edx.devstack.lms:18000"
+      CMS_BASE_URL: "http://tahoe.devstack.studio:18010"
+      APPSEMBLER_EDX_OAUTH_CLIENT_ID: 6f2b93d5c02560c3f93f
+      APPSEMBLER_EDX_OAUTH_CLIENT_SECRET: 2c6c9ac52dd19d7255dd569fb7eedbe0ebdab2db
+      DJANGO_SETTINGS_MODULE: config.settings
+      DJANGO_SECRET_KEY: DUMMY KEY ONLY FOR TO DEVSTACK
+      DJANGO_ALLOWED_HOSTS: '*'
+      OAUTHLIB_INSECURE_TRANSPORT: "true"
+      APPSEMBLER_EDX_API_KEY: test
+      APPSEMBLER_SECRET_KEY: secret_key
+    image: gcr.io/appsembler-tahoe-0/amc
+    ports:
+      - "19000:19000"
+
+  amc-frontend:
+    command: bash -c 'while true; do npm start; sleep 2; done'
+    container_name: tahoe.devstack.amc-frontend
+    # Allows attachment to the LMS service using 'docker attach <containerID>'.
+    stdin_open: true
+    tty: true
+    environment:
+      LMS_BASE: "localhost:18000"
+      PORT: "13000"
+      HOST: 0.0.0.0
+    image: gcr.io/appsembler-tahoe-0/amc-frontend
+    ports:
+      - "13000:13000"
+
   devpi:
     container_name: edx.devstack.devpi
     image: edxops/devpi:hawthorn.master

--- a/provision-tahoe.py
+++ b/provision-tahoe.py
@@ -3,11 +3,10 @@
 This file belongs to `appsembler/devstack` repo, only edit that version.
 Otherwise your changes will be overridden each time devstack is started.
 """
-from os import makedirs, unlink
+from os import makedirs
 from shutil import move
 from path import Path
 from os import symlink
-import json
 
 
 ENVIRONMENT_FILES = [
@@ -22,48 +21,6 @@ ENVS_DIR = SRC_DIR / 'edxapp-envs'
 EDXAPP_DIR = Path('/edx/app/edxapp')
 
 
-def update_tahoe_env_with_tahoe_prefs(filename):
-    """
-    Apply a couple of patches to the lms/cms.env.json files to match Tahoe needs.
-
-    TODO: Move inside the Docker images, via Ansible.
-    """
-    env_file_path = ENVS_DIR / filename
-    with open(env_file_path, 'r') as env_fd:
-        env_object = json.load(env_fd, encoding='utf-8')
-
-        # Theme is enabled by default, can be disabled from within the lms.env.json file
-        env_object.update({
-            'COMPREHENSIVE_THEME_DIRS': ['/edx/src/themes'],
-            'DEFAULT_SITE_THEME': 'edx-theme-codebase',
-            'ENABLE_COMPREHENSIVE_THEMING': True,
-        })
-
-        env_object['REGISTRATION_EXTRA_FIELDS'] = {
-            'city': 'hidden',
-            'country': 'hidden',
-            'gender': 'optional',
-            'goals': 'optional',
-            'honor_code': 'required',
-            'level_of_education': 'optional',
-            'mailing_address': 'optional',
-            'year_of_birth': 'optional',
-        }
-
-        # Disable by default. Can be enabled from within the lms.env.json file
-        env_object['FEATURES'].update({
-            'ENABLE_TIERS_APP': False,  # TODO: Fix the tiers app in devstack
-            'ENABLE_CREATOR_GROUP': True,
-            'DISABLE_COURSE_CREATION': False,
-            'FIGURES_IS_MULTISITE': True,
-        })
-
-        lms_env_encoded = json.dumps(env_object, ensure_ascii=False, indent=4).encode('utf-8')
-
-    with open(env_file_path, 'w') as env_fd:
-        env_fd.write(lms_env_encoded)
-
-
 def move_environment_files_to_host():
     """
     Move the json environment files to the host so they're editable.
@@ -72,7 +29,6 @@ def move_environment_files_to_host():
         container_path = EDXAPP_DIR / filename
         src_path = ENVS_DIR / filename  # The mounted directory in
 
-        newly_copied = False
         if not src_path.exists():
             if container_path.islink():
                 raise Exception(
@@ -81,16 +37,12 @@ def move_environment_files_to_host():
                 )
 
             move(container_path, src_path)
-            newly_copied = True
 
         if src_path.exists():
             if container_path.exists():
                 container_path.unlink()
 
             symlink(src_path, container_path)
-
-        if newly_copied and filename.endswith('.env.json'):
-            update_tahoe_env_with_tahoe_prefs(filename)
 
 
 def main():

--- a/repo.sh
+++ b/repo.sh
@@ -20,6 +20,9 @@ fi
 OPENEDX_GIT_BRANCH=open-release/hawthorn.master
 
 APPSEMBLER_EDX_PLATFORM_BRANCH="appsembler/tahoe/master"
+AMC_BRANCH="omar/docker-amc"  # TODO: Change to `develop` ASAP
+THEME_CODEBASE_BRANCH="hawthorn/master"
+THEME_CUSTOMERS_BRANCH="hawthorn/tahoe"
 
 repos=(
     "https://github.com/edx/course-discovery.git"
@@ -28,7 +31,10 @@ repos=(
     "https://github.com/edx/ecommerce.git"
     "https://github.com/edx/edx-e2e-tests.git"
     "https://github.com/edx/edx-notes-api.git"
-    "https://github.com/appsembler/edx-platform.git"
+    "git@github.com:appsembler/amc.git"
+    "git@github.com:appsembler/edx-theme-codebase.git"
+    "git@github.com:appsembler/edx-theme-customers.git"
+    "git@github.com:appsembler/edx-platform.git"
     "https://github.com/edx/xqueue.git"
     "https://github.com/edx/edx-analytics-pipeline.git"
 )
@@ -86,11 +92,13 @@ _clone ()
             cd ..
         else
             if [ "$name" == "edx-platform" ]; then
-                if [ "${SHALLOW_CLONE}" == "1" ]; then
-                    git clone --single-branch -b ${APPSEMBLER_EDX_PLATFORM_BRANCH} -c core.symlinks=true --depth=1 ${repo}
-                else
-                    git clone --single-branch -b ${APPSEMBLER_EDX_PLATFORM_BRANCH} -c core.symlinks=true ${repo}
-                fi
+                git clone -b ${APPSEMBLER_EDX_PLATFORM_BRANCH} -c core.symlinks=true ${repo}
+            elif [ "$name" == "amc" ]; then
+                git clone -b ${AMC_BRANCH} -c core.symlinks=true ${repo}
+            elif [ "$name" == "edx-theme-codebase" ]; then
+                git clone -b ${THEME_CODEBASE_BRANCH} -c core.symlinks=true ${repo}
+            elif [ "$name" == "edx-theme-customers" ]; then
+                git clone -b ${THEME_CUSTOMERS_BRANCH} -c core.symlinks=true ${repo}
             else
                 if [ "${SHALLOW_CLONE}" == "1" ]; then
                     git clone --single-branch -b ${OPENEDX_GIT_BRANCH} -c core.symlinks=true --depth=1 ${repo}
@@ -115,21 +123,19 @@ _checkout_and_update_branch ()
     fi
 }
 
-# our version to handle the fact that edx-platform needs a
-# different branch, not `master`
+# our version to handle the fact that both edx-platform and AMC need
+# different branches, not `master`
 _appsembler_checkout_and_update_branch ()
 {
     repo="$1"
     if [ "${repo}" == "edx-platform" ]; then
-        
-        GIT_SYMBOLIC_REF="$(git symbolic-ref HEAD 2>/dev/null)"
-        BRANCH_NAME=${GIT_SYMBOLIC_REF##refs/heads/}
-        if [ "${BRANCH_NAME}" == "${APPSEMBLER_EDX_PLATFORM_BRANCH}" ]; then
-            git pull origin ${APPSEMBLER_EDX_PLATFORM_BRANCH}
-        else
-            git fetch origin ${APPSEMBLER_EDX_PLATFORM_BRANCH}
-            git checkout ${APPSEMBLER_EDX_PLATFORM_BRANCH}
-        fi
+        OPENEDX_GIT_BRANCH="${APPSEMBLER_EDX_PLATFORM_BRANCH}" _checkout_and_update_branch
+    elif [ "${repo}" == "amc" ]; then
+        OPENEDX_GIT_BRANCH="${AMC_BRANCH}" _checkout_and_update_branch
+    elif [ "${repo}" == "edx-theme-codebase" ]; then
+        OPENEDX_GIT_BRANCH="${THEME_CODEBASE_BRANCH}" _checkout_and_update_branch
+    elif [ "${repo}" == "edx-theme-customers" ]; then
+        OPENEDX_GIT_BRANCH="${THEME_CUSTOMERS_BRANCH}" _checkout_and_update_branch
     else
         # default to the old behavior
         _checkout_and_update_branch
@@ -156,7 +162,9 @@ reset ()
 
         if [ -d "$name" ]; then
             if [ "$name" == "edx-platform" ]; then
-                cd $name;git reset --hard HEAD;git checkout ${APPSEMBLER_EDX_PLATFORM_BRANCH};git reset --hard origin/${APPSEMBLER_EDX_PLATFORM_BRANCH};git pull;cd "$currDir"                
+                cd $name;git reset --hard HEAD;git checkout ${APPSEMBLER_EDX_PLATFORM_BRANCH};git reset --hard origin/${APPSEMBLER_EDX_PLATFORM_BRANCH};git pull;cd "$currDir"
+            elif [ "$name" == "amc" ]; then
+                cd $name;git reset --hard HEAD;git checkout ${AMC_BRANCH};git reset --hard origin/${AMC_BRANCH};git pull;cd "$currDir"
             else
                 cd $name;git reset --hard HEAD;git checkout open-release/hawthorn.master;git reset --hard origin/open-release/hawthorn.master;git pull;cd "$currDir"
             fi

--- a/tahoe.mk
+++ b/tahoe.mk
@@ -1,20 +1,4 @@
 DEVSTACK_WORKSPACE ?= ..
-THEMES_DIR = $(DEVSTACK_WORKSPACE)/src/themes
-CUSTOMER_THEME_DIR = $(THEMES_DIR)/edx-theme-codebase/customer_specific
-AMC_DIR = $(DEVSTACK_WORKSPACE)/amc
-FIGURES_DIR = $(DEVSTACK_WORKSPACE)/src/figures
-
-UNAME_S := $(shell uname -s)
-LINUX_CMD ?= true
-MACOSX_CMD ?= true
-
-tahoe.os.exec:  ## Executes operating system-specific commands
-ifeq ($(UNAME_S),Linux)
-	bash -c '$(LINUX_CMD)'
-endif
-ifeq ($(UNAME_S),Darwin)
-	bash -c '$(MACOSX_CMD)'
-endif
 
 tahoe.exec.single:  ## Execute a command inside a devstack docker container
 	docker exec -t edx.devstack.$(SERVICE)  \
@@ -24,131 +8,42 @@ tahoe.exec.edxapp:   ## Execute a command in both LMS and Studio (edxapp contain
 	make COMMAND='$(COMMAND)' SERVICE=lms tahoe.exec.single
 	make COMMAND='$(COMMAND)' SERVICE=studio tahoe.exec.single
 
-tahoe.install-pip.edxapp:  ## Install a pip package in both of LMS and Studio
-	make COMMAND='pip install $(PACKAGE)' SERVICE=lms tahoe.exec.single
-	make COMMAND='pip install $(PACKAGE)' SERVICE=studio tahoe.exec.single
+tahoe.chown:  ## Fix annoying docker permission issues
+	sudo chown -R $(USER) $(DEVSTACK_WORKSPACE)
 
-tahoe.chown.dir:  # Fix permissions for a single directory
-	find $(DIRECTORY) \! -user $(USER) -print -exec sudo chown -R $(USER) {} \;
-
-tahoe.chown:  ## Fix an annoying docker permission issue in both `edx-platform` and `src`
-	make DIRECTORY='$(DEVSTACK_WORKSPACE)/edx-platform/' tahoe.chown.dir
-	make DIRECTORY='$(DEVSTACK_WORKSPACE)/src/' tahoe.chown.dir
-
-tahoe.theme.compile:  ## Compile the static assets of the theme
-	make COMMAND='make requirements' tahoe.exec.edxapp
-	make lms-static
-	make studio-static
-
-tahoe.theme.reset:  ## Removes and re-clone the theme with Tahoe branches
+tahoe.provision:  ## Make the devstack more Tahoe'ish
 	make tahoe.chown
-	rm -rf $(THEMES_DIR)
-
-	# Remove and recreated the customer themes css directory
-	sudo rm -rf $(DEVSTACK_WORKSPACE)/src/themes/customer_themes
-	mkdir -p $(DEVSTACK_WORKSPACE)/src/themes/customer_themes/  # Make it writeble for the user
-
-	git clone git@github.com:appsembler/edx-theme-codebase.git $(THEMES_DIR)/edx-theme-codebase
-	cd $(THEMES_DIR)/edx-theme-codebase && git checkout hawthorn/master
-
-	git clone git@github.com:appsembler/edx-theme-customers.git $(CUSTOMER_THEME_DIR)
-	cd $(CUSTOMER_THEME_DIR) && git checkout hawthorn/tahoe
-
-	make tahoe.chown
-
-tahoe.init.provision-script:  ## Execute the `provision-tahoe.py` script in both of LMS and Studio
 	cat $(DEVSTACK_WORKSPACE)/devstack/provision-tahoe.py > $(DEVSTACK_WORKSPACE)/src/provision-tahoe.py
 	make COMMAND='python /edx/src/provision-tahoe.py' tahoe.exec.edxapp
-	# make COMMAND='pip install -e /edx/src/figures' SERVICE=lms tahoe.exec.single
 	rm $(DEVSTACK_WORKSPACE)/src/provision-tahoe.py
-
-tahoe.init:  ## Make the devstack more Tahoe'ish
-	make tahoe.init.provision-script
 	make tahoe.restart || true
 
 tahoe.up:  ## Run the devstack with proper Tahoe settings, use instead of `$ make dev.up`
-	make tahoe.chown
 	make dev.up
 	@sleep 1
-	make tahoe.init
-	# test -d $(FIGURES_DIR) || make tahoe.figures
-	test -d $(CUSTOMER_THEME_DIR) || (make tahoe.theme.reset && make tahoe.theme.compile)
-	test -d $(AMC_DIR) || make amc.reset
-	test -f $(AMC_DIR)/amc/.env || make amc.env-file
+	make tahoe.provision
 	make tahoe.chown
 
 tahoe.envs._delete:  ## Remove settings, in prep for resetting it
 	sudo rm -rf $(DEVSTACK_WORKSPACE)/src/edxapp-envs
 
-tahoe.envs.reset:  ## Reset the JSON envs
-	make down
-	make tahoe.envs._delete
-	make tahoe.up
-
-tahoe.figures:  ## Install Figures
-	make COMMAND='rm -rf /edx/src/figures' SERVICE=lms tahoe.exec.single
-	git clone -b develop git@github.com:appsembler/figures.git $(FIGURES_DIR)
-	make lms-update-db
-
-tahoe.reset.light:  ## Resets the Tahoe settings including a fresh theme copy and new environment files.
-	make down
-	make tahoe.envs._delete
-	make tahoe.theme.reset
-	@sleep 1
-	make tahoe.up
-	make amc.reset
-	make tahoe.theme.compile
-
-tahoe.reset.full:  ## Does a full reset for everything known to devstack. Will loose all git and database changes.
-	make dev.reset
-	make tahoe.reset.light
-
 tahoe.restart:  ## Restarts both of LMS and Studio python processes while keeping the same container
 	make lms-restart
 	make studio-restart
+	make amc-restart
 
-tahoe.amc.oauth-client:  ## Creates the AMC OAuth client in the LMS
-	# Keep in sync with `amc.env.initial` file
+amc.provision:  ## Initializes the AMC
+	docker exec -it tahoe.devstack.amc python manage.py migrate
 	make COMMAND='python manage.py lms create_oauth2_client http://localhost:9000/     http://localhost:9000/oauth2/access_token/ confidential --client_name AMC --client_id 6f2b93d5c02560c3f93f     --client_secret 2c6c9ac52dd19d7255dd569fb7eedbe0ebdab2db --trusted --settings=devstack_docker' SERVICE='lms' tahoe.exec.single
+	docker exec -it tahoe.devstack.amc python manage.py create_devstack_superuser
+	docker exec -it tahoe.devstack.amc-frontend npm install
 
+amc-shell:
+	docker exec -it tahoe.devstack.amc bash
 
-tahoe.amc.superuser: SHELL:=/bin/bash
-tahoe.amc.superuser: ## Creates the devstack admin user
-	@cd $(AMC_DIR)/amc/; \
-	. env/bin/activate; \
-	. .env ; \
-	echo "from django.contrib.auth.models import User; User.objects.filter(username=\"amc\").exists() or User.objects.create_superuser(\"amc\", email=\"amc@example.com\", password=\"amc\");" | python manage.py shell -i python
+amc-frontend-shell:
+	docker exec -it tahoe.devstack.amc-frontend bash
 
-	@echo Created the superuser amc with password=amc successfully
-
-amc.env-file:  ## Removes and uses a fresh copy of the AMC env file
-	rm -f $(AMC_DIR)/amc/.env
-	cp $(DEVSTACK_WORKSPACE)/devstack/amc.env.initial $(AMC_DIR)/amc/.env
-
-amc.reset:  ## Removes and re-initialize AMC
-	rm -rf $(AMC_DIR)
-	git clone git@github.com:appsembler/amc.git $(AMC_DIR)
-
-	make LINUX_CMD='sudo apt install python3 libjpeg8-dev' tahoe.os.exec
-	make MACOSX_CMD='brew install python3 libjpeg zlib' tahoe.os.exec
-
-	cd $(AMC_DIR)/amc/ && virtualenv -p python3 env
-	cd $(AMC_DIR)/amc/ && . env/bin/activate && pip install --upgrade pip
-	cd $(AMC_DIR)/amc/ && . env/bin/activate && pip install -r ../requirements/local.txt
-
-	cd $(AMC_DIR)/frontend/ && npm install || true
-
-	make amc.env-file
-
-	make amc.migrate
-	make tahoe.amc.oauth-client
-	make tahoe.amc.superuser
-
-amc.start.backend:  ## Starts the AMC Django app
-	bash -c 'cd $(AMC_DIR)/amc/ && source env/bin/activate && source .env && python manage.py runserver localhost:19000'
-
-amc.migrate:  ## Migrate the AMC database
-	bash -c 'cd $(AMC_DIR)/amc/ && source env/bin/activate && source .env && python manage.py migrate'
-
-amc.start.frontend:  ## Starts the AMC Frontend
-	bash -c 'cd $(AMC_DIR)/frontend/ && LMS_BASE='localhost:18000' PORT=13000 HOST=localhost npm start'
+amc-restart:
+	docker exec -t tahoe.devstack.amc bash -c 'killall5'
+	docker exec -t tahoe.devstack.amc-frontend bash -c 'killall5'

